### PR TITLE
feat(picker): ability to show a search input with options to filter data

### PR DESCRIPTION
**Pull request by kefahB**
Hello @NathanWalker,

First Happy new year and best luck for 2023 with all and especially with NS ;)

I've added the filtering feature to the picker .. it is handy with large data such as countries list.

### CODE REVIEW

**Comment by kefahB** [comment]
Hi @NathanWalker 

What do you think about this ?

**Comment by NathanWalker** [comment]
This is amazing @kefahB sorry for delayed response, happy new year indeed!

File: package.json [hunk 1]
```diff
@@ -27,10 +29,12 @@
 		"@angular/router": "^15.0.0",
 		"@nativescript/angular": "^15.0.0",
 		"@nativescript/core": "~8.4.0",
+		"@nativescript/nx": "^4.2.0",
```
Comment (NathanWalker) [inline package.json]: This shouldn't be needed here I don't believe? can likely remove this.

File: package.json [hunk 2]
```diff
@@ -27,10 +29,12 @@
...
 		"@nativescript/plugin-tools": "5.0.3",
 		"@nativescript/types": "~8.4.0",
 		"@nativescript/webpack": "~5.0.5",
 		"@ngtools/webpack": "^15.0.0",
+		"@nrwl/js": "^15.4.2",
```
Comment (NathanWalker) [inline package.json]: Don't believe this is needed either, can remove this.

File: package.json [hunk 3]
```diff
@@ -16,6 +16,8 @@
 	},
 	"private": true,
 	"devDependencies": {
+		"@angular-devkit/build-angular": "^15.0.0",
+		"@angular-devkit/schematics": "^15.0.4",
```
Comment (NathanWalker) [inline package.json]: "@angular-devkit/schematics" can be removed as well.

File: package.json (hunk 1 above)
Comment (kefahB) [reply package.json]: For some reason, the repo doesn't build without those changes! may I have a misconfiguration or some think like this.

Comment (NathanWalker) [reply package.json]: Try removing all these that I mentioned and then run: “npm run setup “ to reset the dependencies.

**Comment by kefahB** [comment]
Hi @NathanWalker 

Can you merge this please.

Thanks :)

**Comment by NathanWalker** [comment]
@kefahB will merge Thursday morning and publish a minor bump.

**Comment by NathanWalker** [comment]
Published with 4.1.0

**Pull request merged**